module/VuFind/src/VuFind/Db/Entity/PagesEntityInterface.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * Interface for representing a notifications pages record.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Db_Interface
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Johannes Schultze <schultze@effective-webwork.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Db\Entity;
+
+use DateTime;
+
+/**
+ * Interface for representing a notifications pages record.
+ *
+ * @category VuFind
+ * @package  Db_Interface
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Johannes Schultze <schultze@effective-webwork.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+interface PagesEntityInterface extends EntityInterface
+{
+    /**
+     * Primary key id getter
+     *
+     * @return int|null
+     */
+
+    public function getPrimaryKeyId(): ?int;
```

Comment:
This interface should have getters and setters for all of the fields in the notifications_pages table. It would also make sense for it to be named NotificationsPagesInterface for consistency with the table name, and to differentiate from any other kinds of pages that might show up in the database in the future.

I'm not sure what `getPrimaryKeyId` is for, but if this is an internal Laminas thing that I'm forgetting about, then it definitely should _not_ be part of the interface.

(The purpose of the interface is to define all data fields, and create setters and getters to access them; when we migrate to Doctrine, using magic properties representing the field names will no longer be an option).

Reply:
I think it is a technical id that is returned after a new entry in the database is created. This id has to be set for all broadcasts or pages objects as the page_id, to connect versions of different languages.

I will add setters and getters.

Reply:
Maybe I'm misunderstanding something, but I think your `getPrimaryKeyId` method will return the same value as `getId`, it's just pulling it from a different place. I think it would be good to eliminate this if it's truly redundant, but I'm happy to discuss further or look more carefully if I'm mistaken.